### DISCUSSION TITLE
Login not working as expected?

### DISCUSSION THREAD

**Discussion by HeartlandTechie** [homebridge-simplisafe3/homebridge-simplisafe3#296] [Q&A] [answered]
Trying to get the Oauth login for Simplisafe isn't working as described anymore . . . not sure if it's my combo of computer/etc... or what.  Can someone verify the login procedures are still working like they describe?   I never get a 'mobile URL' to put in?

**Comment by cgmckeever** [comment, marked as answer]
are you in Chrome? You need to scrape it out of developer tools. Some better directions in the README

**Reply by HeartlandTechie** [reply]
I've tried Safari, Chrome . . . it only gives me the tvd based url . . . and then I have to allow it via email to allow?

**Reply by shamoon** [reply]
Yea most recent release doesnt really change anything with auth stuff but of course could be an issue, though Im not seeing any. As noted the readme has more explanation: https://github.com/homebridge-simplisafe3/homebridge-simplisafe3#simplisafe-authentication but yes, you need to approve the login via email and then are taken to the URL you need to copy / paste (in Chrome you arent sent to the URL you have to look in the dev tools).

Let us know if theres an actual problem.

**Reply by HeartlandTechie** [reply]
Yea, it doesn't work right . . . I never get the url to copy/paste . . . could you explain how that works?   I'm on a Mac with Safari 15 if that makes any difference.   I also tried doing it in Chrome.   I have 2Factor turned on with mobile verification also if that makes a difference?

**Reply by shamoon** [reply]
If you can be more specific about whats happening we might be able to help, perhaps even including logs if needed. Its hard without much info to go on. I updated the readme instructions to be even more explicit, but here's how the process works (copypasta'd from readme):

1. Upon clicking the "Launch SimpliSafe Login" button you are taken to the SimpliSafe login page. If you are already logged in this step is skipped.
2. You will be redirected to a page requesting you to approve the login, either via email or 2FA.
3. After approval you will be redirected to a final URL (begins with com.SimpliSafe.mobile://) which you will need to copy & paste back into the Homebridge settings UI. See notes below about using certain browsers / platforms.

If something is different / not working for you would be helpful to get detail.

**Reply by vickersnm** [reply]
Yea I am in this same boat.  I scrubbed developer tools and I still don't see the url you are speaking of.  The issue i believe he is having im having is it goes straight to your home page.  There if you scrub the console there is no url that looks like com.SimpliSafe.mobile://.  I am in Chrome.  The instructions said to go into the console and find it.  Not new to the console and I still don't see it.

**Reply by shamoon** [reply]
What does “it goes to your home page” mean? Guys, details, please…

**Reply by vickersnm** [reply]
.....When you login to Simple Safe, after you have verfied your account it goes straight to your home page/Dashboard/landing page.  This is where you can see all you camera's and such and/or control your system.  Based on your documentation in chrome here we should be able to look at the developer console to find the 
com.SimpliSafe.mobile:// url.  This url is not there.  You can see under java scripts all the Oauth/auth calls but nothing that looks like that url that is requested to be placed in the redirect section.  

URLS you see look like below:
https://webapp.simplisafe.com/
https://cdnjs.cloudflare.com/
https://cdn.optimizely.com/
https://cdn.optimizely.com/
https://bam-cell.nr-data.net/
https://api.simplisafe.com/
https://stats.g.doubleclick.net
https://api.simplisafe.com
https://auth.august.com/

I scrubbed the whole console and no where does it have this phrase "com.SimpliSafe.mobile://".

I tried the other auth urls and no such luck.  I would suspect your code is looking for that specific url but it doesn't seem its provided anymore when looking in the developer console within chrome.

Let me know if you need anything else.  Not sure I can explain home page/dashboard/landing page more since those are pretty standard terms.  If you own the simple safe platform you should see this when you login.  There will be a menu on the left and in the center are three big buttons with off home away.  It's the first page you come to after login.

**Reply by shamoon** [reply]
Yea something is off about that. The URL the settings should launch is

`https://auth.simplisafe.com/authorize?client_id=42aBZ5lYrVW12jfOuu3CQROitwxg9sN5&scope=offline_access%20email%20openid%20https://api.simplisafe.com/scopes/user:platform&response_type=code&redirect_uri=com.simplisafe.mobile%3A%2F%2Fauth.simplisafe.com%2Fios%2Fcom.simplisafe.mobile%2Fcallback&code_challenge_method=S256&code_challenge=__&audience=https://api.simplisafe.com/&auth0Client=__
`

Note the `**redirect_uri=com.simplisafe.mobile%3A%2F%2Fauth.simplisafe.com**` part. That means that after youre authenticated you should be redirected to that address, not webapp.simplisafe.com...

Are you using HOOBS by any chance?

**Reply by vickersnm** [reply]
No, i installed homebridge on one of my local machines, then installed the plugin my self.

**Reply by shamoon** [reply]
Did you see the other persons comment? Tried incognito/private mode? Something doesn’t make sense

**Reply by vickersnm** [reply]
Ok, so good news in chrome i finally got the error page to get the url.  I got it all synced into homebridge.  Just putting this here for reference i had to restart home bridge a few times get finally get the feeds to start coming in from simplisafe.  All is good for now.  Thanks for the help shamoon

**Comment by HeartlandTechie** [comment]
I tried a different approach today . . . I went into incognito mode on Safari, opened my UI for the Homebridge app, it prompted me for my username/password and then when I logged in it sent me a text that I autofilled from my Mac/Safari.  Then it redirected me to the com.simplisafe.mobile://auth.simplisafe.com/ios/com.simplisafe.mobile/callback?code=xxxxxx site that resulted in a error message on screen, but I copied that to the Homebridge plugin and it worked.   Maybe this will help someone else?

**Reply by shamoon** [reply]
Ah, interesting. Thanks for sharing. Im not exactly clear why that situation arises, I am not able to reproduce that but perhaps something about the process above was stored in a cookie from a previous login. Glad to hear its sorted for you 👍

**Comment by RicardoPassavantes** [comment]
I'm trying to do the OAuth process in Safari 15 on Monterey Beta 12.3 and having similar errors

In normal browsing mode, when I click the link in the Settings for the plug-in I get a new tab with a modal error with:

<Safari cannot open the page because the address is invalid.>

When I try from a Private Browsing session I get the authentication prompt, I get a new tab which has the authentication page (for Simplisafe for iOS), I authenticate, and get the 2FA prompt, enter the authenticator code, then when I click on the continue button I get the same modal error:

<Safari cannot open the page because the address is invalid.>

Clicking OK in the error simply closes the error, then the "Continue" button doesn't do anything

I tried it in MS Edge (chromium based) and had similar issues with blank pages, nothing happening.

I had to go into an InPrivate browsing session in Edge, go through the process, get the error - but then the URL was in the console.

Not sure what that tells you, but Safari 15 seems to not like trying to open the special URL, and Private Browsing seems to be required?

**Reply by shamoon** [reply]
I dont usually install beta OS on my primary machine, when its released I can test in Safari but yes, sounds like will require alternative browser / private mode or something. Again, sometimes things change from betas to release. Either way, clearly pretty out of our control but will update docs, etc. when the time comes.

**Comment by shamoon** [comment]
Hi all, if youre interested please check out https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/releases/tag/v1.10.0-beta.3 which includes "automatic" authorization i.e. no browser stuff =)

As its a beta, please ensure you take backups of everything first, just in case.
